Fix LightEditor toggle context scope

ericmehl opened the pull request:
Previously, we were not scoping the viewer context when attempting to toggle a boolean value automatically. Because the `__toggleBoolean()` method uses `fullAttributes()` to determine the new toggle value, it would not take the script context into account. This in turn caused things like the mute toggle to not work if one of a light's attributes did not provide a default value in a script context variable expression.

I think the fix in `LightEditor.py` is pretty straightforward, but I'm not entirely sure about the test case exercising it. I'll make comments inline about what I'm not certain of.

### Checklist ###

- [X] I have read the [contribution guidelines](https://github.com/GafferHQ/gaffer/blob/main/CONTRIBUTING.md).
- [X] I have updated the documentation, if applicable.
- [X] I have tested my change(s) in the test suite, and added new test cases where necessary.
- [X] My code follows the Gaffer project's prevailing coding style and conventions.

ericmehl commented on python/GafferSceneUITest/LightEditorTest.py:
```diff
@@ -656,6 +660,51 @@ def testLightMuteAttribute( toggleCount, toggleLocation, newStates ) :
+
+		editor = GafferSceneUI.LightEditor( script )
+		widget = editor._LightEditor__pathListing
+		editor._LightEditor__settingsNode["in"].setInput( script["custAttr"]["out"] )
```
With this removed, the test will pass _both_ before and after the proposed fix.

With it included, it only passes with the fix.

I expect that without this line, the test would fail both before the fix (and maybe after too?). Where is it getting the context variable `test`?

johnhaddon replied:
> Where is it getting the context variable test?

I don't think it is getting the context variable. Without the `setInput()` call, it's not showing anything at all, so it probably never calls `fullAttributes()`. `_LightEditor__settingsNode` is a bit of a hacky way of doing this though - we should just use the public API and call `editor.setNodeSet( Gaffer.StandardSet( [ script["custAttr"] ) )`.

ericmehl replied:
That makes sense. I changed to using `editor.setNodeSet()` and squashed into https://github.com/GafferHQ/gaffer/pull/5408/commits/d647680ee95b87544412fe504968d68e9d10f69e

ericmehl commented on python/GafferSceneUITest/LightEditorTest.py:
```diff
@@ -633,20 +650,7 @@ def testLightMuteAttribute( toggleCount, toggleLocation, newStates ) :
 			widget = editor._LightEditor__pathListing
-			columns = widget.getColumns()
-
-			if muteIndex is None :
```
Fixed and squashed into https://github.com/GafferHQ/gaffer/pull/5408/commits/d647680ee95b87544412fe504968d68e9d10f69e

ericmehl commented:
I addressed the two comments and squashed it into https://github.com/GafferHQ/gaffer/pull/5408/commits/d647680ee95b87544412fe504968d68e9d10f69e. Assuming CI is happy, ready for a new look.
